src/odemis/gui/util/img.py

```diff
@@ -2621,37 +2621,38 @@ def insert_tile_to_image(tile, ovv):
     return ovv
 
 
-def merge_screen(im, background):
+def merge_screen(ima, imb):
     """ 
-    Merges two images (im and background) into one using the "screen" operator:
+    Merges two images into one using the "screen" operator:
     f(xA,xB) = xA + xB − xA·xB (with values between 0 and 1, with each channel independent)
-    im, background: DataArray with im.shape = background.shape (either RGB or RGBA)
-    returns RGBA DataArray (of the same YX as im, but with always depth=4)
+    ima, imb: DataArray with ima.shape = imb.shape (either RGB or RGBA)
+    returns RGBA DataArray (of the same YX as ima, but with always depth=4)
     """
-    assert im.shape == background.shape, "Images have different shapes."
-    if im.shape[-1] != 3 and im.shape[-1] != 4:
-        raise ValueError("Ovv images have an invalid number of channels: %d" % (im.shape[-1]))
+    if ima.shape[-1] != 3 and ima.shape[-1] != 4:
+        raise ValueError("Ovv images have an invalid number of channels: %d" % (ima.shape[-1]))
+    if ima.shape[:-1] != imb.shape[:-1]:
+        raise ValueError("Images have different shapes: %s != %s" % (ima.shape, imb.shape))
 
-    md = background.metadata.copy()
-    im = format_rgba_darray(im, 255)  # convert to BGRA
-    im.metadata["dc_keepalpha"] = True
-    background = format_rgba_darray(background, 255)
+    md = imb.metadata.copy()
+    ima = format_rgba_darray(ima, 255)  # convert to BGRA
```

Comment:
Why is the function 'format_rgba_darray' not called 'format_rgba_**to**_darray'?

Reply:
Just bad naming. The function actually has a "TODO: get a better name". One day someone should take care of that... It's actually not "to darray", it's "to bgra"!